intelmq/lib/utils.py

```diff
@@ -808,3 +812,51 @@ def file_name_from_response(response: requests.Response) -> str:
     except KeyError:
         file_name = response.url.split("/")[-1]
     return file_name
+
+
+def list_all_bots() -> str:
+    mod = importlib.import_module('intelmq.bots')
+    dirs = ['collectors', 'parsers', 'experts', 'outputs']
+    allowed_types = ['bool', 'NoneType', 'str', 'int']
+    bot_types = [
+        'CollectorBot',
+        'ParserBot',
+        'OutputBot',
+        'Bot'
+    ]
+    root = mod.__path__[0]
+    root_dict = {}
+
+    for d in dirs:
+        root_dict[d.capitalize()] = {}
+        for dirs in os.listdir(os.path.join(root, d)):
+            bot = Path(os.path.join(root, d, dirs))
+            if bot.is_dir() and "__" not in bot.name:
+                for files in os.listdir(bot):
+                    f = Path(files)
+
+                    if f.suffix == '.py' and '__' not in f.name:
+                        mod = importlib.import_module("intelmq.bots.{}.{}.{}".format(d, bot.parts[-1:][0], f.name[0:-3]))
+                        if hasattr(mod, 'BOT'):
+                            name = mod.BOT.__name__
+                            keys = {}
+                            for prop, val in vars(mod.BOT).items():
+                                if prop.startswith("_"):
+                                    continue
+
+                                if type(getattr(mod.BOT, prop)).__name__ not in allowed_types:
+                                    continue
```

Comment:
Maybe its again easier to use a flat list instead of a for loop and if statements with continue:
```
bot = mod.BOT
variables = sorted(name for name, value in inspect.getmembers(bot) if not inspect.ismethod(value) and not inspect.isfunction(value) and not name.startswith('_'))
```
Maybe that's why there is the `allowed_types`?